components/admin-tabs/Applicants.tsx

```diff
@@ -1,35 +1,203 @@
-import React, { useState } from 'react';
+import React, { useContext, useState } from 'react';
 import { ADMIN_TABS } from '../../common/constants';
-import { getAllApplicants } from '../../common/apiClient';
-import useSWR from 'swr';
-import { Button, Table, TablePaginationConfig, TableProps, Tooltip } from 'antd';
-import { ApplicationStatus, Dropdown, User } from '../../common/types';
+import { getAllApplicants, updateApplicantById } from '../../common/apiClient';
+import useSWR, { KeyedMutator } from 'swr';
+import {
+  Table,
+  TablePaginationConfig,
+  TableProps,
+  Form,
+  Select,
+  notification,
+  Tooltip,
+  Button,
+} from 'antd';
+import {
+  ApplicationStatus,
+  Dropdown as DropdownQuestionType,
+  RSVPStatus,
+  ApplicantsApiResponse,
+} from '../../common/types';
 import { Questions } from '../../common/questions';
+import { FormInstance } from 'antd/lib/form';
+import { SelectValue } from 'antd/lib/select';
 import { saveAs } from 'file-saver';
+import { AxiosResponse } from 'axios';
 
-// table columns: name, email, school, application status
-const columns = [
+const { Option } = Select;
+const EditableContext = React.createContext<FormInstance | null>(null);
+type SingleRecordType = ApplicantsApiResponse['data'][0];
```

Comment:
also to explain this: this is just to make all the form stuff generic over the record type, which in this case, is the type `User & { _id: string }` (taken from the ApplicantsApiResponse)